pkgs/development/libraries/audio/cubeb/default.nix

```diff
@@ -0,0 +1,56 @@
+{ lib, stdenv, fetchFromGitHub
+, cmake
+, pkg-config
+, jack2
+, pulseaudio
+, sndio
+, speexdsp
+, lazyLoad ? true
+}:
+
+stdenv.mkDerivation {
+  pname = "cubeb";
+  version = "unstable-2022-10-18";
+
+  src = fetchFromGitHub {
+    owner = "mozilla";
+    repo = "cubeb";
+    rev = "27d2a102b0b75d9e49d43bc1ea516233fb87d778";
+    hash = "sha256-q+uz1dGU4LdlPogL1nwCR/KuOX4Oy3HhMdA6aJylBRk=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+  ];
+
+  buildInputs = [
+    jack2
+    pulseaudio
+    sndio
+    speexdsp
+  ];
+
+  cmakeFlags = [
+    "-DBUILD_SHARED_LIBS=ON"
+    "-DBUILD_TESTS=OFF" # tests require an audio server
+    "-DBUNDLE_SPEEX=OFF"
+    "-DUSE_SANITIZERS=OFF"
+
+    # Whether to lazily load libraries with dlopen()
+    "-DLAZY_LOAD_LIBS=${if lazyLoad then "ON" else "OFF"}"
+  ];
+
+  passthru = {
+    # For downstream users when lazyLoad is true
+    backendLibs = [ jack2 pulseaudio sndio speexdsp ];
```

Comment:
I have a few questions (I know this PR is merged, I was thinking about making one to address these):
- Does `speexdsp` (which is not an audio backend) need to be in `backendLibs` with the other audio backends?
- Isn't `libpulseaudio` enough compared to `pulseaudio`?
- Shouldn't `alsa-lib` also be there? (It's tier 3, but so is `jack2`)

Reply:
Good points. Maybe we should make this customizable, and have a sane default where everything is added.

If you want to open a PR I can review it later.

Reply:
Good poimts! A PR would be nice and we can test there.

Reply:
#200241 (where I also integrate Cubeb better in other emulators)